assets/js/theme/common/carousel.js

```diff
@@ -2,20 +2,44 @@ import 'slick-carousel';
 
 export default function () {
     const $carousel = $('[data-slick]');
-    if ($carousel.length) {
-        const multipleSlides = $carousel[0].childElementCount > 1;
-        $carousel.slick({ dots: multipleSlides });
-    }
+
+    if ($carousel.length === 0) return;
+
+    const multipleSlides = $carousel[0].childElementCount > 1;
+    $carousel.slick({ dots: multipleSlides });
+
+    const $slidesNodes = $('.heroCarousel-slide');
+
+    $slidesNodes.each((index, element) => {
```

Comment:
if you will merge two `each` methods, you will not need `$slidesNodes` variable

```suggestion
    $('.heroCarousel-slide').each((index, element) => {
```